shared/src/business/useCaseHelper/trialSessions/setNoticeOfChangeToRemoteProceeding.js

```diff
@@ -0,0 +1,146 @@
+ * @returns {object} the created trial session
+ */
+exports.setNoticeOfChangeToRemoteProceeding = async (
+  applicationContext,
+  {
+    caseEntity,
+    currentTrialSession,
+    newPdfDoc,
+    newTrialSessionEntity,
+    PDFDocument,
+    userId,
+  },
+) => {
+  const shouldIssueNoticeOfChangeToRemoteProceeding =
+    currentTrialSession.proceedingType ===
+      TRIAL_SESSION_PROCEEDING_TYPES.inPerson &&
+    newTrialSessionEntity.proceedingType ===
+      TRIAL_SESSION_PROCEEDING_TYPES.remote &&
+    caseEntity.status !== CASE_STATUS_TYPES.closed;
+
+  if (shouldIssueNoticeOfChangeToRemoteProceeding) {
+    const trialSessionInformation = {
+      chambersPhoneNumber: newTrialSessionEntity.chambersPhoneNumber,
+      joinPhoneNumber: newTrialSessionEntity.joinPhoneNumber,
+      judgeName: newTrialSessionEntity.judge.name,
+      meetingId: newTrialSessionEntity.meetingId,
+      password: newTrialSessionEntity.password,
+      startDate: newTrialSessionEntity.startDate,
+      startTime: newTrialSessionEntity.startTime,
+      trialLocation: newTrialSessionEntity.trialLocation,
+    };
+
+    const notice = await applicationContext
+      .getUseCases()
+      .generateNoticeOfChangeToRemoteProceedingInteractor(applicationContext, {
+        docketNumber: caseEntity.docketNumber,
+        trialSessionInformation,
+      });
+
+    const docketEntryId = applicationContext.getUniqueId();
+
+    await applicationContext.getPersistenceGateway().saveDocumentFromLambda({
+      applicationContext,
+      document: notice,
+      key: docketEntryId,
+    });
+
+    const noticeOfChangeToRemoteProceedingDocketEntry = new DocketEntry(
+      {
+        date: newTrialSessionEntity.startDate,
+        docketEntryId,
+        documentTitle:
+          SYSTEM_GENERATED_DOCUMENT_TYPES.noticeOfChangeToRemoteProceeding
+            .documentTitle,
+        documentType:
+          SYSTEM_GENERATED_DOCUMENT_TYPES.noticeOfChangeToRemoteProceeding
+            .documentType,
+        eventCode:
+          SYSTEM_GENERATED_DOCUMENT_TYPES.noticeOfChangeToRemoteProceeding
+            .eventCode,
+        isAutoGenerated: true,
+        isFileAttached: true,
+        isOnDocketRecord: true,
+        processingStatus: DOCUMENT_PROCESSING_STATUS_OPTIONS.COMPLETE,
+        signedAt: applicationContext.getUtilities().createISODateString(),
+        trialLocation: newTrialSessionEntity.trialLocation,
+        userId,
+      },
+      { applicationContext },
+    );
+
+    noticeOfChangeToRemoteProceedingDocketEntry.numberOfPages =
+      await applicationContext.getUseCaseHelpers().countPagesInDocument({
+        applicationContext,
+        docketEntryId:
+          noticeOfChangeToRemoteProceedingDocketEntry.docketEntryId,
+      });
+
+    caseEntity.addDocketEntry(noticeOfChangeToRemoteProceedingDocketEntry);
+    const servedParties = aggregatePartiesForService(caseEntity);
+
+    noticeOfChangeToRemoteProceedingDocketEntry.setAsServed(servedParties.all);
```

Comment:
Nit: would it be useful to set the docket entry as served after the persistence call that serves it? just in case that call fails

Reply:
Good idea! We saw it done this way in other places that were doing similar stuff, do you think we should change those calls too? Or maybe we need stick to the scope of this story?

Reply:
Hey actually I just remembered this is not possible to do - `serveNoticesForCase` expects certain served properties to already have been set on the docket entry, and then proceeds to send the service emails. So we MUST call `setAsServed` before calling `serveNoticesForCase`